### ISSUE TITLE
JsonFormat and JsonPropertyOrder not supported for Jackson deserialization

### ISSUE THREAD

**Issue by alexwen** [immutables/immutables#199]
Testing on the latest version, 2.1.0, It appears as though Jackson's JsonFormat and JsonPropertyOrder are not supported:

```
@JsonFormat(shape = JsonFormat.Shape.ARRAY)
@JsonPropertyOrder({"longitude", "latitude", "altitude"})
@JsonInclude(JsonInclude.Include.NON_EMPTY)
@JsonDeserialize(as = ImmutableGeoPoint.class)
@JsonSerialize(as = ImmutableGeoPoint.class)
public abstract class GeoPoint {
```
Previously, in 2.0.17, this could be deserialized from an array: `[1, 2, 3]`. This is no longer the case. Serialization works fine.

Thanks for the help!

**Comment by elucash** [comment]
I'll see what we can do to fix this. Thank you for the bug report!

**Comment by elucash** [comment]
Sorry for the long wait, now it should work, fixed and released in 2.1.1. Thank you and @ldriscoll for PR #206

**Comment by alexwen** [comment]
Thats no problem, thanks for the fix and the great library! I will be testing out soon.
